Fix usage of mac address patterns in schema.py to allow using variables (Closes #238)

lpalgarvio opened the pull request:
- Adds BLANK and VAR patterns
- Patches MAC patterns
- Removes usage of maxLength and minLength in mac and bssid properties

#238

lpalgarvio commented:
tested as such:
```
root@openwisp:~# cp /tmp/schema.py /opt/openwisp2/env/lib/python3.7/site-packages/netjsonconfig/schema.py
root@openwisp:~# systemctl restart supervisor.service 
root@openwisp:~# supervisorctl restart openwisp2
openwisp2: stopped
openwisp2: started
root@openwisp:~# systemctl restart nginx.service
```

pandafy reviewed (changes requested):
Thank you contributing @lpalgarvio! I have left some suggestions below which will improve this PR.

lpalgarvio commented:
> Thank you contributing @lpalgarvio! I have left some suggestions below which will improve this PR.

Cool, ill try these and update as soon as possible :)

lpalgarvio commented:
> Thank you contributing @lpalgarvio! I have left some suggestions below which will improve this PR.

Sorry for the delay! A bit overloaded with OpenWrt projects and other things :)
Confirmed working! Added your commit suggestions.
Do you want me to squash these 3 commits?
@pandafy

lpalgarvio commented:
Also maybe i should rename the commit to add "[fix]" to unbreak the CI checks

pandafy commented on netjsonconfig/schema.py:
```diff
@@ -8,8 +8,11 @@
 
 DEFAULT_FILE_MODE = '0644'
 X509_FILE_MODE = '0600'
+BLANK_PATTERN = '(\s)'
```
We are not using `BLANK_PATTERN` anymore. Is it safe to remove this?

lpalgarvio replied:
indeed, removing

pandafy reviewed (changes requested):
Thanks for following up @lpalgarvio! I have tested your patch with openwisp-controller and these changes work as expected. 

I have asked for one more improvement below. After that we should be able to merge this. Let us know if you'll be able to fix it. 

> Also maybe i should rename the commit to add "[fix]" to unbreak the CI checks

@lpalgarvio you are correct. The QA checks of OpenWISP requires the commit messages in a specific format. Please read the [contribution guidelines](https://openwisp.io/docs/developer/contributing.html#how-to-commit-your-changes-properly). Don't worry about squashing the commits, we'll handle that when we merge the PR.

lpalgarvio commented:
> Thanks for following up @lpalgarvio! I have tested your patch with openwisp-controller and these changes work as expected.
> 
> I have asked for one more improvement below. After that we should be able to merge this. Let us know if you'll be able to fix it.
> 
> > Also maybe i should rename the commit to add "[fix]" to unbreak the CI checks
> 
> @lpalgarvio you are correct. The QA checks of OpenWISP requires the commit messages in a specific format. Please read the [contribution guidelines](https://openwisp.io/docs/developer/contributing.html#how-to-commit-your-changes-properly). Don't worry about squashing the commits, we'll handle that when we merge the PR.

fixed PR title, commit title and description, and because was already rebasing anyways to fix titles, squashed all commits

pandafy reviewed (approved):
Thank you very much for contributing @lpalgarvio! LGTM 👍🏼 

Deferring merge to @nemesisdesign 😄
